hazelcast-sql/src/main/java/com/hazelcast/jet/sql/impl/connector/map/MapIndexScanP.java

```diff
@@ -0,0 +1,450 @@
+import com.hazelcast.jet.core.AbstractProcessor;
+import com.hazelcast.jet.core.Processor;
+import com.hazelcast.jet.core.ProcessorMetaSupplier;
+import com.hazelcast.jet.core.ProcessorSupplier;
+import com.hazelcast.jet.impl.connector.AbstractIndexReader;
+import com.hazelcast.jet.sql.impl.SimpleExpressionEvalContext;
+import com.hazelcast.map.impl.operation.MapFetchIndexOperation.MapFetchIndexOperationResult;
+import com.hazelcast.map.impl.operation.MapFetchIndexOperation.MissingPartitionException;
+import com.hazelcast.map.impl.operation.MapOperationProvider;
+import com.hazelcast.map.impl.proxy.MapProxyImpl;
+import com.hazelcast.nio.ObjectDataInput;
+import com.hazelcast.nio.ObjectDataOutput;
+import com.hazelcast.nio.serialization.IdentifiedDataSerializable;
+import com.hazelcast.query.impl.QueryableEntry;
+import com.hazelcast.query.impl.getters.Extractors;
+import com.hazelcast.spi.impl.InternalCompletableFuture;
+import com.hazelcast.spi.impl.operationservice.Operation;
+import com.hazelcast.sql.impl.exec.scan.MapScanRow;
+import com.hazelcast.sql.impl.exec.scan.index.IndexFilter;
+import com.hazelcast.sql.impl.expression.ExpressionEvalContext;
+import com.hazelcast.sql.impl.expression.predicate.TernaryLogic;
+import com.hazelcast.sql.impl.plan.node.MapIndexScanMetadata;
+
+import javax.annotation.Nonnull;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.PrimitiveIterator;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ExecutionException;
+import java.util.function.Function;
+import java.util.stream.IntStream;
+
+import static com.hazelcast.instance.impl.HazelcastInstanceFactory.getHazelcastInstance;
+import static com.hazelcast.jet.impl.util.Util.getNodeEngine;
+import static com.hazelcast.jet.sql.impl.ExpressionUtil.evaluate;
+import static com.hazelcast.jet.sql.impl.JetSqlSerializerHook.F_ID;
+import static com.hazelcast.jet.sql.impl.JetSqlSerializerHook.IMAP_INDEX_SCAN_PROCESSOR_META_SUPPLIER;
+import static com.hazelcast.jet.sql.impl.JetSqlSerializerHook.IMAP_INDEX_SCAN_PROCESSOR_SUPPLIER;
+import static java.util.Collections.emptyList;
+import static java.util.stream.Collectors.toList;
+
+/*
+Behavior of the processor
+- Converts `IndexFilter` to `IndexIterationPointer[]`.
+- Replaces `ParameterExpression` with `ConstantExpression`
+- initially it assumes that all assigned partitions are local. Sends IndexFetchOp with all local partitions
+- When response is received, saves the new `pointers` to `lastPointers` and
+  sends another one immediately with the new pointers and then emits to processor's outbox
+*/
+
+/**
+ * // TODO: [sasha] detailed description
+ */
+public final class MapIndexScanP extends AbstractProcessor {
+    @SuppressWarnings({"checkstyle:StaticVariableName", "checkstyle:MagicNumber"})
+    static int FETCH_SIZE_HINT = 128;
```

Comment:
Make it `final` and remove suppressions?

Reply:
Thank you for review!